scripts/audit_chains/audit_contracts_setup.js

```diff
@@ -325,139 +328,145 @@ async function main() {
+            "chiado": "scripts/deployment/l2/globals_gnosis_chiado.json",
+            "arbitrumOne": "scripts/deployment/l2/globals_arbitrum_one.json",
+            "arbitrumSepolia": "scripts/deployment/l2/globals_arbitrum_sepolia.json",
+            "optimistic": "scripts/deployment/l2/globals_optimistic_mainnet.json",
+            "optimisticSepolia": "scripts/deployment/l2/globals_optimistic_sepolia.json",
+            "base": "scripts/deployment/l2/globals_base_mainnet.json",
+            "baseSepolia": "scripts/deployment/l2/globals_base_sepolia.json",
+            "celo": "scripts/deployment/l2/globals_celo_mainnet.json",
+            "celoAlfajores": "scripts/deployment/l2/globals_celo_alfajores.json"
+        };
+
+        const providerLinks = {
+            "mainnet": "https://eth-mainnet.g.alchemy.com/v2/" + process.env.ALCHEMY_API_KEY_MAINNET,
+            "goerli": "https://eth-goerli.g.alchemy.com/v2/" + process.env.ALCHEMY_API_KEY_GOERLI,
+            "polygon": "https://polygon-mainnet.g.alchemy.com/v2/" + process.env.ALCHEMY_API_KEY_MATIC,
+            "polygonMumbai": "https://polygon-mumbai.g.alchemy.com/v2/" + process.env.ALCHEMY_API_KEY_MUMBAI,
+            "gnosis": "https://rpc.gnosischain.com",
+            "chiado": "https://rpc.chiadochain.net",
+            "arbitrumOne": "https://arb1.arbitrum.io/rpc",
+            "arbitrumSepolia": "https://sepolia-rollup.arbitrum.io/rpc",
+            "optimistic": "https://optimism.drpc.org",
+            "optimisticSepolia": "https://sepolia.optimism.io",
+            "base": "https://mainnet.base.org",
+            "baseSepolia": "https://sepolia.base.org",
+            "celo": "https://forno.celo.org",
+            "celoAlfajores": "https://alfajores-forno.celo-testnet.org"
+        };
+
+        // Get all the globals processed
+        const globals = new Array();
+        const providers = new Array();
+        for (let i = 0; i < numChains; i++) {
+            const dataJSON = fs.readFileSync(globalNames[configs[i]["name"]], "utf8");
+            globals.push(JSON.parse(dataJSON));
+            const provider = new ethers.providers.JsonRpcProvider(providerLinks[configs[i]["name"]]);
+            providers.push(provider);
+        }
 
-        log = initLog + ", contract: " + "ServiceManagerToken";
-        await checkServiceManager(configs[i]["chainId"], providers[i], globals[i], configs[i]["contracts"], "ServiceManagerToken", log);
+        console.log("\nVerifying deployed contracts setup... If no error is output, then the contracts are correct.");
 
-        log = initLog + ", contract: " + "ServiceRegistryTokenUtility";
-        await checkServiceRegistryTokenUtility(configs[i]["chainId"], providers[i], globals[i], configs[i]["contracts"], "ServiceRegistryTokenUtility", log);
+        for (let i = 0; i < numChains; i++) {
+            // Skip not fully setup networks
+            if (i == 12 || i == 13) {
```

Comment:
We can now do all the chains in one loop, skipping those that are not deployed yet.